Source/GTMSessionFetcher.m

```diff
@@ -3749,21 +3753,30 @@ - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
  * run after the completion handler. Serial dispatch is guaranteed by dispatching to an intermediate
  * serial queue that targets the actual callback queue (which may or may not be serial).
  */
-- (nonnull dispatch_queue_t)serialCallbackQueue {
+- (nonnull dispatch_queue_t)internalCallbackQueue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    return [self serialCallbackQueueUnsynchronized];
+    return [self internalCallbackQueueUnsynchronized];
   }  // @synchronized(self)
 }
 
-- (nonnull dispatch_queue_t)serialCallbackQueueUnsynchronized {
+- (nonnull dispatch_queue_t)internalCallbackQueueUnsynchronized {
   GTMSessionCheckSynchronized(self);
 
-  if (!_serialCallbackQueue) {
-    _serialCallbackQueue = SerialCallbackQueueForTargetQueue(_callbackQueue);
+  if (!_internalCallbackQueue) {
```

Comment:
Since `_internalCallbackQueue` is what we're using for the actual dispatches, the `_callbackQueue`, whatever it is, needs to get assigned to it when it's the main queue. I can move the check into here and just have the serial call do the wrapping explicitly.